inst/include/population_dynamics/fleet/fleet.hpp

```diff
@@ -142,81 +171,134 @@ namespace fims_popdy {
+                    0); /**<model landings numbers at age*/
+            std::fill(landings_weight_at_age.begin(), landings_weight_at_age.end(),
+                    0); /**<model landings weight at age*/
+            std::fill(landings_numbers_at_length.begin(), landings_numbers_at_length.end(),
+                    0); /**<model landings numbers at length*/
+
+            //index
+            std::fill(index_weight.begin(), index_weight.end(),
+                    0); /**<model index of abundance in weight*/
+            std::fill(index_numbers.begin(), index_numbers.end(),
+                    0); /**<model index of abundance in numbers*/
+            std::fill(index_expected.begin(), index_expected.end(),
+                    0); /**<model expected index of abundance*/
+            std::fill(log_index_weight.begin(), log_index_weight.end(),
+                    0); /**<model log of index of abundance in weight*/
+            std::fill(log_index_numbers.begin(), log_index_numbers.end(),
+                    0); /**<model log of index of abundance in numbers*/
+            std::fill(log_index_expected.begin(), log_index_expected.end(),
+                    0); /**<model log of expected index of abundance*/
+            std::fill(index_numbers_at_age.begin(), index_numbers_at_age.end(),
+                    0); /**<model index numbers at age*/
+            std::fill(index_weight_at_age.begin(), index_weight_at_age.end(),
+                    0); /**<model index weight at age*/
+            std::fill(index_numbers_at_length.begin(), index_numbers_at_length.end(),
+                    0); /**<model index numbers at length*/
+
+            //composition
+            std::fill(composition_numbers_at_age.begin(), composition_numbers_at_age.end(),
+                    0); /**<model composition numbers at age*/
+            std::fill(composition_numbers_at_length.begin(), composition_numbers_at_length.end(),
+                    0); /**<model composition numbers at length*/
+            std::fill(proportion_numbers_at_age.begin(), proportion_numbers_at_age.end(),
+                    0); /**<model expected composition proportion numbers at age*/
+            std::fill(proportion_numbers_at_length.begin(), proportion_numbers_at_length.end(),
+                    0); /**<model expected composition proportion numbers at length*/
         }
 
         /**
-         * Evaluate the proportion of catch numbers at age.
+         * Evaluate the proportion of landings numbers at age.
          */
         void evaluate_age_comp() {
             for (size_t y = 0; y < this->nyears; y++) {
                 Type sum = 0.0;
+                //Type sum_obs = 0.0;
                 for (size_t a = 0; a < this->nages; a++) {
                     size_t i_age_year = y * this->nages + a;
-                    sum += this->catch_numbers_at_age[i_age_year];
+                    sum += this->composition_numbers_at_age[i_age_year];
```

Comment:
so, I think i understand now that composition is in numbers and proportions sum to 1, but we always call proportions "comps" so this will probably be confusing to others (not just me :)). why is "composition" needed in the name? Perhaps to denote difference between the population naa and the values scaled by selectivity, q, etc.? why not landings_numbers_at_age_expected and index_numbers_at_age_expected and the like?

Reply:
Getting away from landings at age is why there are now comps. We never actually have landings at age just a composition subsample that has the same selectivity. The goal here was to make the landings, index, and composition calculations independent from each other so that fleets can have any one without the others. That is why it is now composition_numbers_at_age.